### PR TITLE
PWN-5814 - add RoundedDecoration, rounded ripple background

### PR DESCRIPTION
**Pull request by NikolaiEmelianov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-5814

## Description of Work

RoundedDecoration, add `fun View.rippleBackground` for rounded ripple 

https://user-images.githubusercontent.com/121934748/216276631-ff6fe6bd-5681-4143-8fb4-e57269c63bab.mp4

### CODE REVIEW

File: ui-kit/build.gradle [hunk 1]
```diff
@@ -4,8 +4,9 @@ dependencies {
     implementation project(":core")
 
     implementation 'androidx.core:core-ktx:1.9.0'
-    implementation 'androidx.appcompat:appcompat:1.5.1'
+    implementation 'androidx.appcompat:appcompat:1.6.0'
     implementation 'androidx.swiperefreshlayout:swiperefreshlayout:1.1.0'
+    implementation 'androidx.recyclerview:recyclerview:1.2.1'
```
Comment (1sakov) [inline ui-kit/build.gradle:9]: `com.google.android.material:material` should include recycler.

**Review by 1sakov — APPROVED** [review]
lgtm

File: ui-kit/src/main/kotlin/org/p2p/uikit/utils/recycler/RoundedDecoration.kt [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import org.p2p.uikit.utils.shapeBottomRounded
+import org.p2p.uikit.utils.shapeRectangle
+import org.p2p.uikit.utils.shapeRoundedAll
+import org.p2p.uikit.utils.shapeTopRounded
+
+interface RoundItem {
+    fun needDecorate(): Boolean
+}
+
+class RoundedDecoration constructor(roundDp: Float) : RecyclerView.ItemDecoration() {
```
Comment (gslevinkov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/utils/recycler/RoundedDecoration.kt:17]: RoundedItemDecoration

no need for constructor keyword

File: ui-kit/src/main/kotlin/org/p2p/uikit/utils/recycler/RoundedDecoration.kt [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+    private val shapeTopRounded: ShapeAppearanceModel = shapeTopRounded(roundDp)
+    private val shapeRounded: ShapeAppearanceModel = shapeRoundedAll(roundDp)
+    private val shapeBottomRounded: ShapeAppearanceModel = shapeBottomRounded(roundDp)
+    private val shapeRectangle: ShapeAppearanceModel = shapeRectangle()
+
+    private fun RecyclerView.ViewHolder.notNeedDecorate(): Boolean {
```
Comment (gslevinkov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/utils/recycler/RoundedDecoration.kt:24]: noNeedToDecorate

File: ui-kit/src/main/kotlin/org/p2p/uikit/utils/ShapeAppearanceExtensions.kt [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    bottomLeftDp = bottomDp,
+)
+
+fun shapeRoundedAll(
+    cornerSizeDp: Float = 0f,
+) = ShapeAppearanceModel.builder()
+    .setAllCorners(CornerFamily.ROUNDED, cornerSizeDp)
+    .build()
+
+fun shapeRounded(
```
Comment (gslevinkov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/utils/ShapeAppearanceExtensions.kt:111]: please put explicit return types everywhere

**Review by gslevinkov — APPROVED** [review]

**Comment by gslevinkov** [comment]
what's with task in the PR? it's not yours. wrong number?

**Review by kabramovp2p — APPROVED** [review]

File: ui-kit/build.gradle (hunk 1 above)
Comment (NikolaiEmelianov) [reply ui-kit/build.gradle:9]: Снимок экрана 2023-02-02 в 13 19 37

two versions without it 
uikit used old
Снимок экрана 2023-02-02 в 13 19 53

Comment (1sakov) [reply ui-kit/build.gradle:9]: mb we can update material version then?

**Comment by NikolaiEmelianov** [comment]
> 



> what's with task in the PR? it's not yours. wrong number?

@kabramovp2p said. it need for new history and swap

**Pull request merged**